### DISCUSSION TITLE
ec2 describe-availability-zones returns and empty list for any region except the current region

### DISCUSSION THREAD

**Discussion by heavyweight280** [aws/aws-cli#8362] [Q&A] [answered]
Hello all. I am getting an empty list whenever I run 
`aws ec2 describe-availability-zones --filters=Name=region-name,Values=<ANY REGION BUT CURRENT>`
The only way I can see any results is if i also pass `--region=<REGION>`. 

I found this when trying to troubleshoot an identical issue in terraform when using this to get availability zones form a region:
`data "aws_availability_zones" "available" {
    filter {
        name =  "region-name"
        values = ["ANY-REGION-THAT-ISNT-CURRENT"]
    }
}`

This will also return an empty list unless the filter is for the currently used region. So the filter doesn't seem to do anything helpful.
Is this the expected behavior? Thank you in advance.

**Comment by RyanFitzSimmonsAK** [comment, marked as answer]
Hi @heavyweight280, thanks for reaching out. DescribeAvailabilityZones only returns availability zones in the region of the request, and by default, only returns availability zones that are available to you. So yes, getting an empty list from a filter on other regions would be expected behavior.

@github-actions proposed answer
